src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -0,0 +1,465 @@
+    // you may not use this file except in compliance with the License.
+    // You may obtain a copy of the License at
+    //
+    //     http://www.apache.org/licenses/LICENSE-2.0
+    //
+    // Unless required by applicable law or agreed to in writing, software
+    // distributed under the License is distributed on an "AS IS" BASIS,
+    // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    // See the License for the specific language governing permissions and
+    // limitations under the License.
+
+    // Generated code. DO NOT EDIT!
+
+    @join import : common.api.imports
+        using {@import.fullName};
+    @end
+
+    namespace {@common.api.packageName}
+    {
+        {@settings(common.api, common.settings)}
+
+        {@abstractClient(common.api, common.settings)}
+
+        {@implClient(common.api, common.settings)}
+        
+        {@partialPageStreaming(common.api, common.settings)}
+    }
+{@""}
+@end
+
+@private settings(api, settings)
+    /// <summary>
+    /// Settings for a <see cref="{@api.name}"/>.
+    /// </summary>
+    public sealed partial class {@settings.name} : ServiceSettingsBase
+    {
+        /// <summary>
+        /// Get a new instance of the default <see cref="{@settings.name}"/>.
+        /// </summary>
+        /// <returns>
+        /// A new instance of the default <see cref="{@settings.name}"/>.
+        /// </returns>
+        public static {@settings.name} GetDefault() => new {@settings.name}();
+
+        /// <summary>
+        /// Constructs a new <see cref="{@settings.name}"/> object with default settings.
+        /// </summary>
+        public {@settings.name}() { }
+        
+        private {@settings.name}({@settings.name} existing) : base(existing)
+        {
+            GaxPreconditions.CheckNotNull(existing, nameof(existing));
+            @join call : settings.callSettings
+                {@call.methodName}Settings = existing.{@call.methodName}Settings?.Clone();
+            @end
+        }
+        
+        @join retry : settings.retryCodesDefinitions
+            /// <summary>
+            /// The filter specifying which RPC <see cref="StatusCode"/>s are eligible for retry
+            /// for "{@retry.name}" <see cref="{@api.name}"/> RPC methods.
+            /// </summary>
+            /// <remarks>
+            @if retry.codes
+               /// The eligible RPC <see cref="StatusCode"/>s for retry for "{@retry.name}" RPC methods are:
+               /// <list type="bullet">
+               @join code : retry.codeNames
+                   /// <item><description><see cref="StatusCode.{@code}"/></description></item>
+               @end
+               /// </list>
+            @else
+                /// There are no RPC <see cref="StatusCode"/>s eligible for retry for "{@retry.name}" RPC methods.
+            @end
+            /// </remarks>
+            public static Predicate<RpcException> {@retry.name}RetryFilter { get; } =
```

Comment:
I would argue that composed identifiers are already opaque. I don't know what the form of retry.name is here; I can only tell that RetryFilter is being added onto it. 

I'm trying to push for composing the strings in one place so that it is easier to predict where to look for something - either it is a direct string (find it in the template), or it is composed (find it in the surface namer).